content/rma_intro.tex

```diff
@@ -16,33 +16,59 @@
 for block, scalar, and block-strided put and get communication. 
-Such type-generic routines are supported for the ``standard \ac{RMA} types’’
-identified in Table \ref{stdrmatypes}.
+Such type-generic routines are supported for the
+\newtext{\Celev{}
+``standard integer types'' (\Celev{} \S6.2.5.7),
+``real floating types'' (\Celev{} \S6.2.5.10),
+``exact-width integer types'' (\Celev{} \S7.20.1.1),
+``integer types capable of holding object pointers'' (\Celev{} \S7.20.1.4),
+and the integer types defined in stddef.h (\Celev{} \S7.19),
+which are collectively called the} ``standard \ac{RMA} types''
+\newtext{and are listed} \oldtext{identified} in Table \ref{stdrmatypes}.
+\newtext{Implementations may optionally support additional types.}
 
 \begin{table}[h]
   \begin{center}
     \begin{tabular}{|l|l|}
       \hline
-      \TYPE & \TYPENAME\\
-      \hline
-      float & float\\
-      \hline
-      double & double\\
-      \hline
-      long double & longdouble\\
-      \hline
-      char & char\\
-      \hline
-      short & short\\
-      \hline
-      int & int\\
-      \hline
-      long & long\\
-      \hline
-      long long & longlong\\
-      \hline
+      \TYPE              & \TYPENAME  \\ \hline
+      float              & float      \\ \hline
+      double             & double     \\ \hline
+      long double        & longdouble \\ \hline
+      char               & char       \\ \hline
+%\newtext {
+      signed char        & schar      \\ \hline
+%}
+      short              & short      \\ \hline
+      int                & int        \\ \hline
+      long               & long       \\ \hline
+      long long          & longlong   \\ \hline
+%\newtext {
+      bool \footnotemark & bool       \\ \hline
+      unsigned char      & uchar      \\ \hline
+      unsigned short     & ushort     \\ \hline
+      unsigned int       & uint       \\ \hline
+      unsigned long      & ulong      \\ \hline
+      unsigned long long & ulonglong  \\ \hline
+      int8\_t            & int8       \\ \hline
+      int16\_t           & int16      \\ \hline
+      int32\_t           & int32      \\ \hline
+      int64\_t           & int64      \\ \hline
+      uint8\_t           & uint8      \\ \hline
+      uint16\_t          & uint16     \\ \hline
+      uint32\_t          & uint32     \\ \hline
+      uint64\_t          & uint64     \\ \hline
+      size\_t            & size       \\ \hline
+      ptrdiff\_t         & ptrdiff    \\ \hline
+      wchar\_t           & wchar      \\ \hline
+      intptr\_t          & intptr     \\ \hline
+      uintptr\_t         & uintptr    \\ \hline
+%}
     \end{tabular}
     \caption{Standard \ac{RMA} Types and Names}
     \label{stdrmatypes}
   \end{center} 
 \end{table}
+
+\footnotetext{\newtext{In order to maintain compatibility between C and C++,
+the C interface requires the \VAR{bool} macro as defined in stdbool.h.}}
```

Comment:
I suggest that we introduce a C99 baseline requirement for the C bindings and change this footnote to indicate that implementations may choose to use the type ```_Bool``` to *avoid* a dependency on ```stdbool.h```.  I have no idea how to support a ```bool``` type (nor do I know how to support a ```complex``` type) portably prior to C99.  We can further indicate that implementations may optionally choose to support earlier versions of the C language by disabling bindings for types not defined prior to C99.

If we want ```bool``` and ```complex``` to work in C++, then we should really define proper C++ bindings.  As defined, C++ users are calling C functions and would need to supply arguments compatible with the C types.